core/src/test/scala/unit/kafka/log/LogTest.scala

```diff
@@ -2363,6 +2370,8 @@ class LogTest {
 
   private def testDegenerateSplitSegmentWithOverflow(segmentBaseOffset: Long, records: List[MemoryRecords]): Unit = {
     val segment = LogTest.rawSegment(logDir, segmentBaseOffset)
+    Log.offsetIndexFile(logDir, segmentBaseOffset).createNewFile()
+    Log.timeIndexFile(logDir, segmentBaseOffset).createNewFile()
```

Comment:
Probably add a comment why we need to explicitly create those files.

Reply:
Done.